mm2src/database/stats_swaps.rs

```diff
@@ -243,6 +251,51 @@ fn insert_stats_taker_swap_sql_init(swap: &TakerSavedSwap) -> Option<(&'static s
     Some((INSERT_STATS_SWAP_ON_INIT, params))
 }
 
+fn update_stats_swap_db(
+    ctx: &MmArc,
+    maker_coin_usd_price: &str,
+    taker_coin_usd_price: &str,
+    uuid: &str,
+) -> SqlResult<()> {
+    debug!("Updating swap coins usd price for {} to the SQLite database", uuid);
+    let conn = ctx.sqlite_connection();
+    let params = [uuid, maker_coin_usd_price, taker_coin_usd_price];
+
+    conn.execute(UPDATE_STATS_SWAP, &params).map(|_| ())
+}
+
+pub async fn update_stats_swap_coins_price_to_db(
+    ctx: &MmArc,
+    swap: &SavedSwap,
+    base_usd_price: BigDecimal,
+    rel_usd_price: BigDecimal,
+) -> Result<(), String> {
+    match &swap {
+        SavedSwap::Maker(maker) => {
+            if let Err(e) = update_stats_swap_db(
+                ctx,
+                &base_usd_price.to_string(),
+                &rel_usd_price.to_string(),
+                &maker.uuid.to_string(),
+            ) {
+                error!("Error updating maker stats_swaps db -> {}", e);
```

Comment:
The `update_stats_swap_coins_price_to_db` function either should not return anything (so we log the error) or we need to return the error here.
I think we need to return an error here. It can be `SqlError` rather than `String`.